basics/timelock-contract/src/TimelockContract.cs

```diff
@@ -0,0 +1,131 @@
+        }
+
+        public override Empty SetPendingAdmin(SetPendingAdminInput input)
+        {
+            // Assert(Context.Sender == Context.Self, "No permission");
+            Assert(State.Admin.Value == Context.Sender, "No permission");
+            State.PendingAdmin.Value = input.PendingAdmin;
+            Context.Fire(new NewPendingAdmin
+            {
+                NewPendingAdmin_ = input.PendingAdmin
+            });
+            return new Empty();
+        }
+
+        public override Empty QueueTransaction(TransactionInput input)
+        {
+            Assert(Context.Sender == State.Admin.Value, "No permission");
+            Assert(input.Eta >= Context.CurrentBlockTime.AddDays(State.Delay.Value), "Estimated execution block must satisfy delay");
+            Hash txnHash = HashHelper.ComputeFrom(input);
+            State.TransactionQueue[txnHash] = Context.Sender;
+            Context.Fire(new QueueTransaction
+            {
+                TxnHash = txnHash,
+                Target = input.Target,
+                Amount = input.Amount,
+                Signature = input.Signature,
+                Data = input.Data,
+                Eta = input.Eta
+            });
+            return new Empty();
+        }
+
+        public override Empty CancelTransaction(TransactionInput input)
+        {
+            Assert(Context.Sender == State.Admin.Value, "No permission");
+            Hash txnHash = HashHelper.ComputeFrom(input);
+            State.TransactionQueue.Remove(txnHash);
+            Context.Fire(new CancelTransaction
+            {
+                TxnHash = txnHash,
+                Target = input.Target,
+                Amount = input.Amount,
+                Signature = input.Signature,
+                Data = input.Data,
+                Eta = input.Eta
+            });
+            return new Empty();
+        }
+
+        public override Empty ExecuteTransaction(TransactionInput input)
+        {
+            Assert(Context.Sender == State.Admin.Value, "No permission");
+            Hash txnHash = HashHelper.ComputeFrom(input);
+            Assert(State.TransactionQueue[txnHash] != null, "executeTransaction: Transaction hasn't been queued");
+            Assert(Context.CurrentBlockTime >= input.Eta, "executeTransaction: Transaction hasn't surpassed time lock");
+            Assert(Context.CurrentBlockTime <= input.Eta.AddDays(TimelockContractConstants.GRACE_PERIOD), "executeTransaction: Transaction is stale");
+
+            Address from = State.TransactionQueue[txnHash];
+            State.TokenContract.Value = Context.GetContractAddressByName(SmartContractConstants.TokenContractSystemName);
+            State.TokenContract.TransferFrom.Send(new TransferFromInput
+            {
+                From = from,
+                To = input.Target,
+                Amount = (long) input.Amount,
+                Symbol = TimelockContractConstants.SYMBOL,
+                Memo = "Execution"
+            });
```

Comment:
We want to be able to execute any arbitrary transaction instead of the specific `TransferFrom`. Check our `Context.SendInline()` and Ethereum's `call` and `delegatecall`. https://mirror.xyz/frimoldi.eth/5RObAeE1sPNeEzyGU9gU0olBtVJqnj0MYkJCVtVNGV0

Reply:
Modified.